ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -65,30 +74,39 @@ private[apiserver] final class RateLimitingInterceptor(metrics: Metrics, config:
       None
     } else {
       (for {
-        _ <- metricOverloaded(fullMethodName, apiServices, config.maxApiServicesQueueSize)
+        _ <- queueOverloaded(apiServices, config.maxApiServicesQueueSize)
+        _ <- queueOverloaded(indexDbThreadpool, config.maxApiServicesIndexDbQueueSize)
         _ <- metricOverloaded(
-          fullMethodName,
-          indexDbThreadpool,
-          config.maxApiServicesIndexDbQueueSize,
+          metricDescription = "Number of active streams",
+          name = metrics.daml.lapi.streams.activeName,
+          value = metrics.daml.lapi.streams.active.getCount + 1, // Once created
+          limit = config.maxStreams,
         )
       } yield ()).fold(Some.apply, _ => None)
     }
   }
 
-  private def metricOverloaded(
-      fullMethodName: String,
+  private def queueOverloaded(
       count: InstrumentedCount,
       limit: Int,
   ): Either[String, Unit] = {
-    val queued = count.queueSize
-    if (queued > limit) {
-      val rpcStatus =
-        s"""
-           | The ${count.name} queue size ($queued) has exceeded the maximum ($limit).
-           | The rejected call was $fullMethodName.
-           | Api services metrics are available at ${count.prefix}.
-          """.stripMargin
+    metricOverloaded(
+      metricDescription = s"${count.name} queue size",
+      name = count.prefix,
+      value = count.queueSize,
+      limit = limit,
+    )
+  }
 
+  private def metricOverloaded(
```

Comment:
I think the return types should stay the same as it makes the composition work.  It may make more sense to assert that things are under limit as opposed to over limit though.  (e.g. `memoryUnderLimit`).